Basket-Donation-Payments/Basket-Donation-Payments/View Controllers/Payment Flow/PaymentFormViewController.m

```diff
@@ -75,6 +59,31 @@ - (void)pay {
     }];
 }
 
+- (void)completePayment {
+    //FIXME: Show loading refreshing control
+    BasketTransaction *basketTx = [BasketTransaction new];
+    basketTx.basketRecipient = self.basket;
+    basketTx.madeByUser = [User currentUser];
+    basketTx.totalAmount = self.totalAmount;
+    [self.basket.allTransactions addObject:basketTx];
+    self.basket.featuredValueDict[@"numberOfDonations"] = [[NSNumber alloc] initWithLong:self.basket.allTransactions.count];
+
+    int sumFeaturedVal = 0;
+    for (NSNumber *val in self.basket.featuredValueDict.allValues) {
+        sumFeaturedVal += [val intValue];
```

Comment:
Not clear what you're counting here, `basket.featuredValueDict.allValues` means all the values of the dictionary, for all keys. It's not clear that these are always ints and that it makes sense to just sum all these things together. Making this dictionary a model class would help clarify the logic too.